packages/react-components/src/components/SendBox.tsx

```diff
@@ -186,13 +196,28 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
   const [textValue, setTextValue] = useState('');
   const [textValueOverflow, setTextValueOverflow] = useState(false);
 
+  /* @conditional-compile-remove(file-sharing) */
+  const [showFileUploadsPendingError, setShowFileUploadsPendingError] = useState(false);
```

Comment:
I haven't looked at the code below yet, but I'll bet managing this state is complex. Can you extract this into a React component that can encapsulate the logic?

Reply:
That way, you won't need so many conditional compilation directives either. You can simply add that component in the send box conditionally (but define the component unconditionally).

Reply:
I see that you already have `SendBoxErrorBar`. So I guess, I'm wondering if we can somehow move the logic for `fileUploadPendingErrorTimeoutRef` inside that component.

Something like:
* Each time user clicks the send button, if there is a pending upload, render `SendBoxErrorBar` with the *current timestamp* (So in this component, we'll need state to track the last time send was clicked)
* Inside `SendBoxErrorBar`, have the logic to return `<></>` if it's been long enough since the last click (last click is provided via prop).

All of this can be unconditional. The only conditionally compiled bit will be the computation of whether there is any pending upload (see my comment below as well)

Reply:
@PorterNan and you both have good suggestions around moving this logic into a different component.
Instead of modifying `SendBoxErrorBar` component (which can be used in the future for other scenarios as well), I would like to introduce a new component `SendBoxErrors` that encapsulates all the complex logic in SendBox